helm/flux-app/templates/_helpers.tpl

```diff
@@ -55,3 +55,23 @@ app.kubernetes.io/instance: {{ .Release.Name | quote }}
 {{- define "controllerVolumeName" -}}
 {{- printf "%s-controller-%s-volume" (include "name" .) .volumeName -}}
 {{- end -}}
+
+{{- define "appPlatform.twoStepInstall" -}}
```

Comment:
I give it that it looks ugly at the first glance, yet it does the good job of enabling the internal upgrade for the app only when it runs in App Platform ecosystem, and only when this ecosystem runs supported Chart Operator. This is mostly to retain faulty behaviour for vanila environments, or old Chart Operators.

Other thing is, in this form it is not portable. Copying it is error prone so my idea would be to extract this functionality to a separate Helm Chart, and import it just like we import libraries in the code. Such techniques are already used by upstream, for example, Bitnami Helm Chart often import a `common` Helm Chart, see [here](https://github.com/bitnami/charts/blob/5278e0000ee42dfb5d39df3d385882a15dbffea9/bitnami/argo-cd/Chart.yaml#L10), that does provide only the necessary functions, see [here](https://github.com/bitnami/charts/tree/master/bitnami/common/templates). This way there is no need to copy paste them to each Helm Chart which is significant win for the cases like this one.